src/lightning_app/utilities/imports.py

```diff
@@ -17,6 +17,23 @@
 from typing import List, Union
 
 from lightning_utilities.core.imports import module_available
+from packaging.requirements import Marker, Requirement
+
+try:
+    from importlib import metadata
+except ImportError:
+    # Python < 3.8
+    import importlib_metadata as metadata  # type: ignore
+
+
+def _get_extras(extras: str = "cloud") -> List[str]:
+    """Get the list of installable packages in the given extras."""
+    from lightning_app import __package_name__
+
+    extras = f"app-{extras}" if __package_name__ == "lightning" else extras
+    requirements = {r: Requirement(r) for r in metadata.requires(__package_name__)}
+    marker = Marker(f'extra == "{extras}"')
+    return [r.split(";")[0].strip() for r, req in requirements.items() if str(req.marker) == str(marker)]
```

Comment:
```suggestion
    return [r.name for r, req in requirements.items() if str(req.marker) == str(marker)]
```

Reply:
Not equivalent unfortunately, `r.name` is just the plain package name but we need the version spec. as well